shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -733,58 +842,110 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 }
 
 - (void)showTextInput {
-  NSAssert([UIApplication sharedApplication].keyWindow != nullptr,
+  UIWindow* keyWindow = [UIApplication sharedApplication].keyWindow;
+  NSAssert(keyWindow != nullptr,
            @"The application must have a key window since the keyboard client "
            @"must be part of the responder chain to function");
   _activeView.textInputDelegate = _textInputDelegate;
-  [_inputHider addSubview:_activeView];
-  [[UIApplication sharedApplication].keyWindow addSubview:_inputHider];
+  if (![_activeView isDescendantOfView:_inputHider]) {
+    [_inputHider addSubview:_activeView];
+  }
+  [keyWindow addSubview:_inputHider];
   [_activeView becomeFirstResponder];
 }
 
 - (void)hideTextInput {
   [_activeView resignFirstResponder];
-  [_activeView removeFromSuperview];
   [_inputHider removeFromSuperview];
 }
 
 - (void)setTextInputClient:(int)client withConfiguration:(NSDictionary*)configuration {
-  NSDictionary* inputType = configuration[@"inputType"];
-  NSString* keyboardAppearance = configuration[@"keyboardAppearance"];
-  if ([configuration[@"obscureText"] boolValue]) {
-    _activeView = _secureView;
+  NSArray* fields = configuration[@"fields"];
+  NSString* clientUniqueId = _uniqueIdFromDictionary(configuration);
+  bool isSecureTextEntry = [configuration[@"obscureText"] boolValue];
+
+  if (fields == nil) {
+    _activeView = isSecureTextEntry ? _nonAutofillSecureInputView : _nonAutofillInputView;
+    [FlutterTextInputPlugin setupInputView:_activeView WithConfiguration:configuration];
+
+    if (![_activeView isDescendantOfView:_inputHider]) {
+      [_inputHider addSubview:_activeView];
+    }
   } else {
-    _activeView = _view;
+    NSAssert(clientUniqueId != nil, @"The client's unique id can't be null");
+    for (FlutterTextInputView* v in _inputViews)
+      [v removeFromSuperview];
+    for (UIView* subview in [_inputHider subviews])
+      [subview removeFromSuperview];
+
+    [_inputViews removeAllObjects];
+
+    for (NSDictionary* field in fields) {
+      FlutterTextInputView* newInputView = [[FlutterTextInputView alloc] init];
+      newInputView.textInputDelegate = _textInputDelegate;
+      [_inputViews addObject:newInputView];
```

Comment:
This is a memory leak, you aren't releasing `newInputView`.  You could also `autorelease` it if you want.